### DISCUSSION TITLE
hc-login for home appliance simulator ?

### DISCUSSION THREAD

**Discussion by zibous** [hcpy2-0/hcpy#80] [Q&A]
Would it be possible to use the home appliance simulators with hc-login?

Or is there another solution ?

```
haid = None
email = sys.argv[1]
password = sys.argv[2]
haid = sys.argv[3]

simulate=False
if not (haid is None):
   simulate=True
```

see: https://developer.home-connect.com/simulator

**Comment by Meatballs1** [comment]
I'm not familiar with it? What would the goal be? There would be no local devices for it to talk to as they are simulated?

**Reply by zibous** [reply]
> There would be no local devices for it to talk to as they are simulated?

With the home appliance simulator you can simulate appliances without them being available locally. This would make it easier to test the functions of an appliance (e.g. dishwasher door opening, starting a program, sequence, etc.).


![simulator](https://github.com/user-attachments/assets/bab5c5e2-650b-48f8-8168-16ec94a64643)

siehe: https://developer.home-connect.com/simulator

**Comment by adec** [comment]
Unfortunately I don't think this would work as the application simulator only works for the public cloud APIs (which are not used by hcpy) and doesn't act like a local device. Apart from downloading the device information hcpy uses the local unpublished private APIs instead.
